snuba/state/cache/redis/backend.py

```diff
@@ -283,6 +283,43 @@ def build_notify_queue_key(task_ident: str) -> str:
         else:
             raise ValueError("unexpected result from script")
 
+    def __get_value_with_simple_readthrough(
+        self,
+        key: str,
+        function: Callable[[], TValue],
+        record_cache_hit_type: Callable[[int], None],
+        timeout: int,
+        timer: Optional[Timer] = None,
+    ) -> TValue:
+        result_key = self.__build_key(key)
+
+        cached_value = self.__client.get(result_key)
+        if timer is not None:
+            timer.mark("cache_get")
+        metric_tags = timer.tags if timer is not None else {}
+
+        if cached_value is not None:
+            record_cache_hit_type(RESULT_VALUE)
+            return self.__codec.decode(cached_value)
+        else:
+            try:
+                value = self.__executor.submit(function).result(timeout)
```

Comment:
I used the concurrent.futures module for function timeout.

Reply:
I don't think it's as necessary here because there's no notification queue. I think it's fine to just run the query synchronously. That's what we do now when we `short_circuit` the cache.

Reply:
Ok. I will update to a synchronous call.

Reply:
While debugging the code, I still opt to maintain the timeout. 

This is because setting the timeout value too high could result in the thread being held up, waiting for the completion of the clickhouse query.
https://github.com/getsentry/snuba/blob/master/snuba/clickhouse/native.py#L382

Reply:
Clickhouse will reject the query if it takes more than 30 seconds as well. So we shouldn't end up actually running a query longer than 30 seconds either way. I tend to agree with Volo: let's remove this added complexity. If we find that Clickhouse is holding threads longer than 30 seconds we can add it back.

Reply:
Ok. I will be updating code to synchronous call.